orbital-camera.ts

```diff
@@ -0,0 +1,253 @@
+ *  
+ * @remarks
+ * The component works using mouse or touch. Therefor it does not work in VR.
+ */
+export class OrbitalCamera extends Component {
+    static TypeName = 'orbital-camera';
+
+    @property.int()
+    mouseButtonIndex = 0;
+
+    @property.float(5)
+    radial = 5;
+
+    @property.float()
+    minElevation = 0.0;
+
+    @property.float(89.99)
+    maxElevation = 89.99;
+
+    @property.float()
+    minZoom = 0.01;
+
+    @property.float(10.0)
+    maxZoom = 10.0;
+
+    @property.float(0.5)
+    xSensitivity = 0.5;
+
+    @property.float(0.5)
+    ySensitivity = 0.5;
+
+    @property.float(0.02)
+    zoomSensitivity = 0.02;
+
+    @property.float(0.9)
+    decelerationFactor = 0.9;
+
+    private mouseDown: boolean = false;
+    private origin = [0, 0, 0];
+    private azimuth = 0;
+    private polar = 45;
+
+    private initialPinchDistance: number = 0;
+    private touchStartX: number = 0;
+    private touchStartY: number = 0;
+
+    private azimuthSpeed: number = 0;
+    private polarSpeed: number = 0;
+
+    start(): void {
+        this.object.getPositionWorld(this.origin);
+        this.updateCamera();
+    }
+
+    onActivate(): void {
+        const canvas = this.engine.canvas;
+
+        canvas.addEventListener('mousemove', this.onMouseMove);
+        if (this.mouseButtonIndex === 2) {
+            canvas.addEventListener('contextmenu', preventDefault, {passive: false});
+        }
+        canvas.addEventListener('mousedown', this.onMouseDown);
+        canvas.addEventListener('mouseup', this.onMouseUp);
+        canvas.addEventListener('wheel', this.onMouseScroll, {passive: false});
+
+        canvas.addEventListener('touchstart', this.onTouchStart, {passive: false});
+        canvas.addEventListener('touchmove', this.onTouchMove, {passive: false});
+        canvas.addEventListener('touchend', this.onTouchEnd);
+    }
+
+    onDeactivate(): void {
+        const canvas = this.engine.canvas;
+
+        canvas.removeEventListener('mousemove', this.onMouseMove);
+        if (this.mouseButtonIndex === 2) {
+            canvas.removeEventListener('contextmenu', preventDefault);
+        }
+        canvas.removeEventListener('mousedown', this.onMouseDown);
+        canvas.removeEventListener('mouseup', this.onMouseUp);
+        canvas.removeEventListener('wheel', this.onMouseScroll);
+
+        canvas.removeEventListener('touchstart', this.onTouchStart);
+        canvas.removeEventListener('touchmove', this.onTouchMove);
+        canvas.removeEventListener('touchend', this.onTouchEnd);
+
+        // Reset state to make sure nothing gets stuck
+        this.mouseDown = false;
+        this.initialPinchDistance = 0;
+        this.touchStartX = 0;
+        this.touchStartY = 0;
+
+        this.azimuthSpeed = 0;
+        this.polarSpeed = 0;
+    }
+
+    update(): void {
+        if (!this.mouseDown) {
+            // Apply deceleration only when the user is not actively dragging
```

Comment:
We use `/* */` everywhere, a bit different than most codebase I guess but now we are too deep in :D

Reply:
I'm not a fan, but I changed it for consistency. I'll look into adding ESLint rules to enforce these kinds of rules in future.